initrd/etc/functions

```diff
@@ -337,6 +337,97 @@ update_checksums()
 	return $rv
 }
 
+print_tree() {
+	find ./ ! -path './kexec*' -print0 | sort -z
+}
+
+# Escape zero-delimited standard input for use in shell.
+# These characters are passed verbatim: a-zA-Z0-9,._+:@%/-
```

Comment:
Since this is no longer trying to create shell-safe output (which is fine, makes sense for whiptail), I think we can allow more characters through to make the output more readable.

I think we'd be fine to escape 00-1f (controls), # (escape char itself), \ (interpreted by whiptail), 7f (del).  Continuing to use #n#r#t#v#b## for those chars is good IMO, hex for everything else.

In particular leaving space alone would make files with spaces more readable.  I don't think escaping other shell chars like "'(){}[], etc. has any benefit.

I think # was a good choice for the escape char.  It's unfortunate that we have to invent another kind of escaping and hope that the user understands it, but # is odd enough not just in file names but any sort of terminal output that I think it's OK.  I liked what less did with other control chars (inverted colors, some unusual character) but your point about `\b` is good, I hadn't checked that one specifically, better safe than sorry.